plugins/k8saudit/plugin.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+
+	"github.com/alecthomas/jsonschema"
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk"
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk/plugins"
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk/plugins/extractor"
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk/plugins/source"
+	"github.com/falcosecurity/plugins/plugins/k8saudit/pkg/k8saudit"
+)
+
+type K8SAuditPluginConfig struct {
+	SSLCertificate string `json:"sslCertificate" jsonschema:"description=The SSL Certificate to be used with the HTTPS Webhook endpoint (Default: /etc/falco/falco.pem)"`
+	MaxEventBytes  uint64 `json:"maxEventBytes"  jsonschema:"description=Max size in bytes for an event JSON payload (Default: 1048576)"`
+}
+
+type K8SAuditPlugin struct {
+	plugins.BasePlugin
+	config    K8SAuditPluginConfig
+	extractor k8saudit.AuditEventExtractor
+}
+
+func (k *K8SAuditPlugin) Info() *plugins.Info {
+	return &plugins.Info{
+		ID:          1,
+		Name:        "k8saudit",
```

Comment:
I had a conversation with @leogr and @Issif some week ago about this. The idea was to have a `k8saudit` plugin which is a 1:1 porting of the legacy Falco implementation. Then, the we can implement different flavors such as `eks-k8saudit`, `aks-k8saudit`, and so on. All of the flavors might want to reuse the `k8saudit.*` package definitions to implement the extraction capabilities of the plugins, so that we have the shared extraction logic all in one place and differ only on the way events are sourced. WDYT?